### DISCUSSION TITLE
Export the SHELL(I/O) of a text work session

### DISCUSSION THREAD

**Discussion by compsystems** [wxMaxima-developers/wxmaxima#1743] [Ideas]
Hello

Electronic translation from Spanish

The Maxima interface is a SHELL with numbered inputs and outputs. I think it would be very important for MAXIMA to import a script in linear text format

for example

(%i1) (-3)^2
(%i2) -3^2

(%i3) -(3^2)
(%i4) solve((x^2-1) = -3)

for which MAXIMA is required to export a shell-like text file, it can have the extension .smx, this file would also export the output

(%i1) (-3)^2
(%o1) 9

(%i2) -3^2
(%o2) -9

(%i3) -(3^2)
(%o3) -9

(%i4) solve((x^2-1) = -3)
(%04)[x = (isquare root(2)),x = ((−i)square root(2))]

Of course, when importing this file, the results or outputs are ignored.

The student can use this file without using maxima to write expressions and their possible solutions, and then compare them with the computational results.

It can also be used by the teacher to bring a series of scripts to their students and perform them in class with MAXIMA.

and finally upload a MAXIMA script in a printed or digital document, since currently it is necessary to copy and paste the inputs and outputs that MAXIMA returns line by line or take screenshots, but the latter do not allow copying the input expressions

**Comment by gunterkoenigsmann** [comment]
If the file's name ends in `.out` I believe wxMaxima will know that you want it to do.

**Reply by compsystems** [reply]
Hello
the SIMPLE-GUI version allows exporting inputs and outputs of a work session.
Menu: edit, Save the terminal in a file (*.out) but not the GUI for Maxima version
